meilisearch-lib/src/index_controller/snapshot.rs

```diff
@@ -0,0 +1,257 @@
+//.times(uuids_num)
+//.returning(move |_, _| Box::pin(ok(())));
+
+//let dir = tempfile::tempdir_in(".").unwrap();
+//let handle = Arc::new(index_handle);
+//let update_handle =
+//UpdateActorHandleImpl::<Vec<u8>>::new(handle.clone(), dir.path(), 4096 * 100).unwrap();
+
+//let snapshot_path = tempfile::tempdir_in(".").unwrap();
+//let snapshot_service = SnapshotService::new(
+//uuid_resolver,
+//update_handle,
+//Duration::from_millis(100),
+//snapshot_path.path().to_owned(),
+//"data.ms".to_string(),
+//);
+
+//snapshot_service.perform_snapshot().await.unwrap();
+//}
+
+//#[actix_rt::test]
+//async fn error_performing_uuid_snapshot() {
+//let mut uuid_resolver = MockUuidResolverHandle::new();
+//uuid_resolver
+//.expect_snapshot()
+//.times(1)
+////abitrary error
+//.returning(|_| Box::pin(err(UuidResolverError::NameAlreadyExist)));
+
+//let update_handle = MockUpdateActorHandle::new();
+
+//let snapshot_path = tempfile::tempdir_in(".").unwrap();
+//let snapshot_service = SnapshotService::new(
+//uuid_resolver,
+//update_handle,
+//Duration::from_millis(100),
+//snapshot_path.path().to_owned(),
+//"data.ms".to_string(),
+//);
+
+//assert!(snapshot_service.perform_snapshot().await.is_err());
+////Nothing was written to the file
+//assert!(!snapshot_path.path().join("data.ms.snapshot").exists());
+//}
+
+//#[actix_rt::test]
+//async fn error_performing_index_snapshot() {
+//let uuid = Uuid::new_v4();
+//let mut uuid_resolver = MockUuidResolverHandle::new();
+//uuid_resolver
+//.expect_snapshot()
+//.times(1)
+//.returning(move |_| Box::pin(ok(HashSet::from_iter(Some(uuid)))));
+
+//let mut update_handle = MockUpdateActorHandle::new();
+//update_handle
+//.expect_snapshot()
+////abitrary error
+//.returning(|_, _| Box::pin(err(UpdateActorError::UnexistingUpdate(0))));
+
+//let snapshot_path = tempfile::tempdir_in(".").unwrap();
+//let snapshot_service = SnapshotService::new(
+//uuid_resolver,
+//update_handle,
+//Duration::from_millis(100),
+//snapshot_path.path().to_owned(),
+//"data.ms".to_string(),
+//);
+
+//assert!(snapshot_service.perform_snapshot().await.is_err());
+////Nothing was written to the file
+//assert!(!snapshot_path.path().join("data.ms.snapshot").exists());
+//}
+
+//#[actix_rt::test]
+//async fn test_loop() {
+//let mut uuid_resolver = MockUuidResolverHandle::new();
+//uuid_resolver
+//.expect_snapshot()
+////we expect the funtion to be called between 2 and 3 time in the given interval.
+//.times(2..4)
+////abitrary error, to short-circuit the function
+//.returning(move |_| Box::pin(err(UuidResolverError::NameAlreadyExist)));
+
+//let update_handle = MockUpdateActorHandle::new();
+
+//let snapshot_path = tempfile::tempdir_in(".").unwrap();
+//let snapshot_service = SnapshotService::new(
+//uuid_resolver,
+//update_handle,
+//Duration::from_millis(100),
+//snapshot_path.path().to_owned(),
+//"data.ms".to_string(),
+//);
+
+//let _ = timeout(Duration::from_millis(300), snapshot_service.run()).await;
+//}
+//}
```

Comment:
What is this?
We should uncomment it or delete it I guess but not let it commented :thinking:

Reply:
I commented them out for now, since they require rewrite, I'll do it later, as I mentionned in the pr message

Reply:
Yes sorry I noticed later that a lot of tests were commented out and that it was written in the PR :+1: